### ISSUE TITLE
Improve speed operating on many files: run `each` groups in parallel?

### ISSUE THREAD

**Issue by alexkreidler** [nushell/nushell#4111]
### Related problem

I ran a command to find the average size of a `description` field in every JSON file in a subdirectory. It took a somewhat long time (12s):
```
benchmark { ls **/*.json | each { open $it.name | get description | str length } | math avg }
───┬────────────────────────
 # │       real time        
───┼────────────────────────
 0 │ 12sec 773ms 849us 81ns 
───┴────────────────────────
``` 

### Describe the solution you'd like

There were 22713 JSON files processed, but it would be great if this could run a bit faster, and improve NuShell's speed when operating on many files in general.
Since I haven't benchmarked this, I don't know what could be a primary resource hog. I did notice however that a single `nu` thread went to 100% CPU for a while.

If you haven't already considered it, parallelizing the execution of `each` could be very valuable. It could be available as an option for `each` or an entire other command like `parallel`. `each group/window` probably could also be parallelized, by prepping the data to pass in and keeping track of the order of the outputs. I'm a total beginner to Nu so I have no idea how complex the implementation of this would be.

Many thanks!

**Comment by jntrnr** [comment]
in the upcoming engine, we now support `par-each`, which will do what you want. I'll mark this so we can confirm after that gets merged into the main nushell code.

**Comment by hustcer** [comment]
> in the upcoming engine, we now support `par-each`, which will do what you want. I'll mark this so we can confirm after that gets merged into the main nushell code.

When can we use `engine-q` in `nushell`? Is there an approximate plan? even if it is just to month

**Comment by jntrnr** [comment]
This is now supported in nushell's latest main, using `par-each`.
